roles/imageset_mirroring/tasks/main.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: Mirror operators using oc-mirror
+  block:
+
+    - name: Validate im_task parameter
+      ansible.builtin.assert:
+        that:
+          - im_task is defined
+          - im_task in ['latest', 'all', 'custom']
+        fail_msg: "im_task must be set to one of: 'latest', 'all', 'custom'."
+        success_msg: "Valid mirror task selected: {{ im_task }}"
+
+    - name: Validate common required parameters
+      ansible.builtin.assert:
+        that:
+          - im_target is defined
+          - im_target | length > 0
+        fail_msg: "im_target is required and must be a non-empty string."
+
+    - name: Validate parameters for latest and all tasks
```

Comment:
you could avoid these specific details by leaving them in argument_specs, that way, it will not fail in the main, but it will on the call of the particular task selected. That is, if the im_task selected is custom and the im_custom_config is not defined, then it will fail. Is all about argument_specs.